src/buffer/out/textBuffer.cpp

```diff
@@ -1183,14 +1183,14 @@ void TextBuffer::ClearScrollback(const til::CoordType start, const til::CoordTyp
         return;
     }
 
+    ClearMarksInRange(til::point{ 0, start }, til::point{ _width, start + height });
+
     if (height <= 0)
     {
         _decommit();

```

Comment:
I still wonder if we need to do something special here. Height 0 means “clear all,” but your mark clearing code probably just takes the coordinates as-is and will in this case clear 0 lines worth of marks? Maybe I still don’t understand?

Reply:
huh? if we decommit the buffer, then we'll delete all the rows. Like, straight up deallocate all of them. Doesn't matter in that case if `ClearMarksInRange` no-op'd because it cleared 0 rows worth of marks. all the rows would then get deallocated (and when they get re-allocated, they'll be reallocated without any mark data)

Reply:
unless I'm totally just misunderstanding you

Reply:
Oh, I think I'm still in the mindset where the mark data lives somewhere _other_ than the rows (in an aside table or something). Thanks!

Reply:
Hah! Same. 😄

Reply:
Don’t we still need to clear the correct *range* of marks though? That was 50% of that change!